spec/queue_spec.rb

```diff
@@ -0,0 +1,103 @@
+class MyQueue
+
+  def initialize
+    @collection = []
+  end
+
+  def dequeue
+    dq_item = @collection[0]
+
+    @collection = @collection.length == 1 ? [] : create_new_queue
+
+    dq_item
+  end
+
+  def empty?
+    @collection.length.zero?
+  end
+
+  def enqueue(item)
+    @collection[entry_point] = item
+  end
+
+  def peek
+    @collection[0]
+  end
+
+
+  private
+
+  def create_new_queue
```

Comment:
Instead of creating a new array, set the first element to `nil` and shift the rest of the elements forward by 1 position.
`@collection[i] = @collection[i+1]`